frontend/src/libs/packages/store/middlewares/middlewares.ts

```diff
@@ -0,0 +1,21 @@
+import { type AnyAction, type Middleware } from '@reduxjs/toolkit';
+import { isRejected } from '@reduxjs/toolkit';
+
+import { NotificationService } from '#libs/packages/services/notification-service.js';
+import { type AppDispatch } from '#libs/packages/store/store.js';
+import { showErrorNotification } from '#slices/notifications/actions.js';
+
+const handleError: Middleware = (api) => {
+  return (next) => {
+    return (action: AnyAction) => {
+      const dispatch: AppDispatch = api.dispatch;
+      if (isRejected(action) && action.error.message) {
+        dispatch(showErrorNotification(action.error.message));
+        NotificationService.error(action.error.message);
+      }
```

Comment:
1. next is out dispatch
2. we should call Notification service inside action
3. let's rename notifications slice to app slice, and method to notify (it will accept not only errors
```suggestion
const handleError: Middleware = () => {
  return (next: AppDispatch) => {
    return (action: AnyAction) => {
      if (isRejected(action) && action.error) {
        dispatch(appActions.notify(action.error.message ?? 'Something went wrong'));
      }
```